src/typeahead/typeahead-window.ts

```diff
@@ -56,13 +56,13 @@ export class NgbTypeaheadWindow {
    */
   @Output('select') selectEvent = new EventEmitter();
 
-  getActive() { return this.results[this._activeIdx]; }
+  getActive() { return this.results[this.activeIdx]; }
 
-  markActive(_activeIdx: number) { this._activeIdx = _activeIdx; }
+  markActive(activeIdx: number) { this.activeIdx = activeIdx; }
```

Comment:
Actually, I think I understood "exported" wrong. Do you mean "exported from the NgModule", "exported using an exportAs attribute in the component", or both? In fact I concentrated on the two first rules, and I have a hard time understanding the rationale and the end goal behind two last ones:

- if a directive is not exported all public methods called from a template need to be annotated with @internal in JSDoc
- if a directive is exported all public methods not meant to be part of the public API need to be made private or annotated with @internal in JSDoc if used from templates.